solid_utils/sampling.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import math
+import warnings
+from .variogram import remove_trend as rt
+def load_geo(attr_geo):
+    """This program calculate the coordinate of the geocoded files 
+    and perform coordinate transformation from longitude and latitude to local coordinate in kilometers.
+    
+    Parameters:
+    geo_attr:attribute of the geocoded data
+
+    Returns:
+    X:coordinates in east direction in km
+    Y:coordinates in north direction in km
+    """
+    
+    X0=float(attr_geo['X_FIRST'])
+    Y0=float(attr_geo['Y_FIRST'])
+    X_step=float(attr_geo['X_STEP'])
+    Y_step=float(attr_geo['Y_STEP'])
+    length=int(attr_geo['LENGTH'])
+    width=int(attr_geo['WIDTH'])
+    
+    # 
+    Y_local_end_ix = math.floor(length/2)
+    Y_local_first_ix = -(length-Y_local_end_ix-1)
+    X_local_end_ix = math.floor(width/2)
+    X_local_first_ix = -(width-X_local_end_ix-1)
+    
+    Y_origin = Y0+Y_step*(-Y_local_first_ix)
+    
+    X_step_local = math.cos(math.radians(Y_origin))*X_step*111.1
+    Y_step_local = Y_step*111.1
```

Comment:
I assume you are using an Earth radius value to get this 111.1 km per deg, could you please add a comment here for the description?

Reply:
Sure, I will add. Here I assume 1) the earth is a sphere with a radius; 2) the distance of one longitude degree at all latitudes is the same as the one at the central latitude. Both of them will introduce bias especially when the scene is large.

I don't know if there are any better ways to do such transform. If so, please tell me. Thanks.